src/providers/goDaddy.ts

```diff
@@ -0,0 +1,16 @@
+import fetch from '../helpers/httpRequest'
+import { getData } from '../api/lib/data'
+
+export const getDomains = () => {
+  const serviceKeys: any = getData('serviceKeys')
+  const { service, value: key } = serviceKeys.find(
+    (el: { key: string }) => el.key === 'GD_Key'
+  )
+  const { value: secret } = serviceKeys.find(
+    (el: { key: string }) => el.key === 'GD_Secret'
+  )
+  const url = `${service}/v1/domains?statuses=ACTIVE`
```

Comment:
The logic will change with every providers. It's hard coded within each provider files.
Like you said it will change for every services then I have no clue how to do a generic function

Reply:
your logic is good, I misunderstood. Didn't see your file name `goDaddy.ts`. I thought i was `index.ts`. sorry